persistence/javadsl/src/main/scala/com/lightbend/lagom/internal/javadsl/persistence/ReadSideActor.scala

```diff
@@ -93,15 +108,24 @@ private[lagom] class ReadSideActor[Event <: AggregateEvent[Event]](
           config.randomBackoffFactor
         ) { () =>
           val handler: ReadSideProcessor.ReadSideHandler[Event] = processorFactory().buildHandler()
-          val futureOffset: Future[Offset]                      = handler.prepare(tag).toScala
+          val futureOffset: Future[LagomOffset]                 = handler.prepare(tag).toScala
 
           scaladsl.Source
-            .fromFuture(futureOffset)
+            .future(futureOffset)
```

Comment:
This was already implemented in https://github.com/lagom/lagom/pull/2614 but that PR was wrongfully targeting `master`. When forward merging `1.6.x` into `master` this will CONFLICT.